shared/lib/approval.ts

```diff
@@ -0,0 +1,40 @@
+import { ApprovalController } from '@metamask/approval-controller';
+import { ApprovalType } from '@metamask/controller-utils';
+import { DIALOG_APPROVAL_TYPES } from '@metamask/snaps-rpc-methods';
+import { providerErrors } from '@metamask/rpc-errors';
+///: BEGIN:ONLY_INCLUDE_IF(keyring-snaps)
+import { SNAP_MANAGE_ACCOUNTS_CONFIRMATION_TYPES } from '../constants/app';
+///: END:ONLY_INCLUDE_IF
+
+export function rejectAllApprovals(approvalController: ApprovalController) {
+  const approvalRequestsById = approvalController.state.pendingApprovals;
+  const approvalRequests = Object.values(approvalRequestsById);
+
+  for (const approvalRequest of approvalRequests) {
+    const { id, type } = approvalRequest;
+
+    switch (type) {
+      case ApprovalType.SnapDialogAlert:
+      case ApprovalType.SnapDialogPrompt:
+      case DIALOG_APPROVAL_TYPES.default:
+        approvalController.accept(id, null);
+        break;
+
+      case ApprovalType.SnapDialogConfirmation:
+        approvalController.accept(id, false);
+        break;
```

Comment:
Ideally we would be able to call `deleteInterface` here too, otherwise this looks great!

Reply:
Done, and from popup close also.